2002-05-09  Jason Merrill  <jason@redhat.com>

	* g++.dg/ext/asm2.C: New test.

--- gcc/testsuite/g++.dg/ext/asm2.C.jj	Wed May 15 13:51:22 2002
+++ gcc/testsuite/g++.dg/ext/asm2.C	Sun May 12 11:25:45 2002
@@ -0,0 +1,12 @@
+// Bug: in a template, we forgot that this was a simple asm, and decided
+// that %edi was a malformed operand specifier.
+
+template <class T> class  I {
+public:
+ void f() { asm ("# mov %edi, %esi" ); }
+};
+
+int main () {
+  I<int> x;
+  x.f();
+}
